### PR TITLE
Change Settings UI -> Distance to recalculate

### PR DESCRIPTION
**Pull request by nazar-kutz**

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/OsmandSettings.java [hunk 1]
```diff
@@ -3467,6 +3467,7 @@ protected boolean setValue(Object prefs, String val) {
 	}
 
 	public final CommonPreference<Float> ROUTE_RECALCULATION_DISTANCE = new FloatPreference("routing_recalc_distance", 0.f).makeProfile();
+	public final CommonPreference<Boolean> USE_ROUTE_RECALCULATION_DISTANCE = new BooleanPreference("use_routing_recalc_distance", false).makeProfile();
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/OsmandSettings.java]: USE_ROUTE_RECALCULATION_DISTANCE it should be reverted, new setting is not needed same as changes to RoutingHelper it only possibly breaks existing functionality.

File: OsmAnd/src/net/osmand/plus/routing/RoutingHelper.java [hunk 2]
```diff
@@ -442,10 +442,12 @@ private Location setCurrentLocation(Location currentLocation, boolean returnUpda
 				if (allowableDeviation == 0) {
 					allowableDeviation = getDefaultAllowedDeviation(settings, route.getAppMode(), posTolerance);
 				}
+				boolean isRecalculationActivated = allowableDeviation > 0 &&
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/routing/RoutingHelper.java]: USE_ROUTE_RECALCULATION_DISTANCE it should be reverted, new setting is not needed same as changes to RoutingHelper it only possibly breaks existing functionality.

**Comment by sonora** [comment]
What are we doing here? We already have a profile-dependent setting for this.

Is this meant to be a bug fix for something?

Rather than introducing yet another user-accessible option, as if this was a user decision and  'matter of taste', I think our task is to provide an algorithm to determine the value in the background, that's what makes a dumb app a progressive app :)

That 'algorithm' can simply be a constant each for each profile. Or it could be something more elaborate, but then we would have to think about which scenarios or deficiencies we are trying to improve, and which parameters an information we have for this.

**Comment by nazar-kutz** [comment]
In this case, this is just an UI improvement.
Additional variable was added to separate the setting state (on / off) and its numerical value.

**Pull request closed without merging**